### DISCUSSION TITLE
can't use -dPrinted option of GS while converting PDF to JPEG

### DISCUSSION THREAD

**Discussion by hardikradixweb** [ImageMagick/ImageMagick#4002] [Help]
When I am trying covert PDF to JPG using following convert command it is not converting pdf as expected, some of the elements(which is under no print layer) of PDF is not visible in generated JPG file.

I was looking for direct option in ImageMagick to display elements which are under no print layer but I didn't found so I have tried GS(Ghostscript) option. When I am using `-dPrinted=false` directly with GS(Ghostscript) command it is giving expected output.

### used commands:

Not working with convert: `convert sample.pdf sample.jpg`

Working with GS: ` gs -dBATCH -dNOPAUSE -dPrinted=false '-sDEVICE=pngalpha'  '-sOutputFile=sample.jpg' sample.pdf
`
This sample PDF have two image one is under no print layer and another without it [sample.pdf](https://github.com/ImageMagick/ImageMagick/files/6907340/sample.pdf)

**This is expected output:**

![expected_output](https://user-images.githubusercontent.com/152806/127649542-7b14b43c-b614-43b7-973c-d03236e5dea9.jpg)

----------------------------------------------------------------------------------------------------------------------------------------------

**This is actual output:**

![actual_output](https://user-images.githubusercontent.com/152806/127649555-f40a47a6-b481-406e-a357-48ac6290050a.jpg)

created above command from help given for pdf with cropbox "`-define pdf:use-cropbox=true`" under [Imagemagick Formats](https://imagemagick.org/script/formats.php#supported) documents.

### used versions:

```
convert --version
Version: ImageMagick 7.0.8-63 Q16 x86_64 2019-12-26 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): cairo fontconfig freetype jng jpeg lcms png rsvg x zlib
```

```
gs -v
GPL Ghostscript 9.26 (2018-11-20)
Copyright (C) 2018 Artifex Software, Inc.  All rights reserved.
```

**Comment by snibgo** [comment]
ImageMagick (IM) does not provide access to all the GS command-line options. You could do one of these:

1. Create an entry on delegates.xml that does what you want. Give it some unique prefix such as "MyPdf". Then you can convert pdf files with command like "magick MyPdf:abc.pdf out.png".

2. Use GS directly.

**Reply by hardikradixweb** [reply]
For some reasons I can't make entry on delegates.xml and use GS command directly as I am using imagemagick as lambda layer in aws lambda. Is there any other way I can convert pdf to jpg with all elements in it without avoiding no print layer elements?

**Comment by snibgo** [comment]
With your PDF, I get the "expected output" with IM v7.1.0-4 and GS v9.19.

> For some reasons I can't make entry on delegates.xml and use GS command directly...

I know nothing about aws lambda, sorry.

In general, you can create a delegates.xml under your home directory. This gives extra entries, that can be used in addition to those in the system-wide delegates.xml. See https://imagemagick.org/script/resources.php

**Reply by hardikradixweb** [reply]
yes but I think imagemagick should adopt '-dprinted' option of GS same as -`define pdf:use-cropbox=true` or -`define pdf:use-trimbox=true` mentioned under [Supported Image Formats](https://imagemagick.org/script/formats.php#supported).

It is newly introduced from GS v9.26 I guess, I have checked with IM v7.1.0-4 and GS v9.27 it is not working but I have checked with IM v7.1.0-4 and GS v9.20 and it working.

Is there anyway I can report this to imagemagick team to adopt in new version as it is needed while converting pdf to image.

**Comment by snibgo** [comment]
So the change is in GS, and you want IM to have a "-define" to suit. That seems a reasonable request. A developer may respond.

**Reply by hardikradixweb** [reply]
yes, you are right, I have posted issue under IM issues.
